File: 프로그래머스/1/42862. 체육복/체육복.cs
using System;

public class Solution 
{
    public int solution(int n, int[] lost, int[] reserve) 
    {
        int[] count = new int[n];
        for(int i=0;i<n;i++)
        {
            count[i] = 1;
        }
        for(int i=0;i<lost.Length;i++)
        {
            count[lost[i]-1]--;
        }
        for(int i=0;i<reserve.Length;i++)
        {
            count[reserve[i]-1]++;
        }
        
        for(int i=0;i<n;i++)
        {
            if(count[i] == 0)
            {
                if(i - 1 > - 1 && i + 1 < n)
                {
                    if(count[i-1] == 2)
                    {
                        count[i-1] = 1;
                        count[i] = 1;
                    }
                    else if(count[i+1] == 2)
                    {
                        count[i+1] = 1;
                        count[i] = 1;
                    }
                    else
                    {
                        continue;
                    }
                }
                else if(i == 0)
                {
                    if(count[i+1] == 2)
                    {
                        count[i+1] = 1;
                        count[i] = 1;
                    }
                    else
                    {
                        continue;
                    }
                }
                else if(i == n - 1)
                {
                    if(count[i-1] == 2)
                    {
                        count[i-1] = 1;
                        count[i] = 1;
                    }
                    else
                    {
                        continue;
                    }
                }
            }
        }
        int answer = 0;
        for(int i=0;i<n;i++)
        {
            if(count[i] > 0)
            {
                answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/42862. 체육복/체육복.kt
class Solution {
fun solution(n: Int, lost: IntArray, reserve: IntArray): Int {
    var answer = 0
    var len = 0
    var lostLen = 0
    val numArray = IntArray(n){1}
    var lost = lost.sorted()
    var reserve = reserve.sorted()

    for(i in 1..n){
        if(len >= reserve.size){
            len = 0
        }
        if(lostLen >= lost.size){
            lostLen = 0
        }
        if(reserve[len] == i){
            numArray[reserve[len] - 1] += 1

            len++
        }
        if(lost[lostLen] == i){
            numArray[lost[lostLen] - 1] -= 1
            lostLen++
        }
    }

    println(numArray.contentToString())
    for(i in 0 until numArray.size){
        if(i == 0){
            if(numArray[i] == 2 && numArray[1] == 0){
                numArray[i] -= 1
                numArray[1] += 1
            }
            continue
        }
        if(i == numArray.size-1){
            if(numArray[i] == 2 && numArray[i - 1] == 0){
                numArray[i] -= 1
                numArray[i - 1] += 1
            }
            continue
        }

        if(numArray[i] == 2 && numArray[i - 1] == 0){
            numArray[i] -= 1
            numArray[i - 1] += 1
        }else if(numArray[i] == 2 && numArray[i + 1] == 0){
            numArray[i] -= 1
            numArray[i + 1] += 1
        }
    }
    println(numArray.contentToString())
    answer = numArray.filter { it >= 1 }.size


    return answer
}
}